dali/ft/daftprogress.cpp

```diff
@@ -88,15 +88,19 @@ void DaftProgress::onProgress(unsigned __int64 sizeDone, unsigned __int64 totalS
         unsigned secsLeft = (unsigned)(timeLeft * cycleToNanoScale /1000000000);
         char temp[20];
         formatTime(temp, secsLeft);
-        displayProgress((unsigned)(sizeDone*100/totalSize), secsLeft, temp, 
-                        sizeDone/scale,totalSize/scale,scaleUnit, 
+
+        unsigned percentDone = (unsigned)(totalSize ? (sizeDone*100/totalSize) : 0);
+
+        displayProgress(percentDone, secsLeft, temp,
+                        sizeDone/scale, totalSize/scale, scaleUnit,
                         (unsigned)(msGone ? (sizeDone-startSize)/msGone : 0),
                         (unsigned)(recentTimeDelta ? recentSizeDelta / recentTimeDelta : 0), numNodes, numReads, numWrites);
 
         if (sizeDone == totalSize)
         {
             formatTime(temp, (unsigned)(msGone/1000));
-            displaySummary(temp, (unsigned)((totalSize - startSize)/msGone));
+            unsigned kbPerSecond = (unsigned)(msGone ? ((totalSize - startSize)/msGone) : 0);
```

Comment:
If msGone == 0 then perhaps kbPerSecond should be not 0 but 1 or something else > 0 ?

Reply:
Perhaps msGone should be set as a minimum to 1 above - see line 82.
if (msGone == 0) msGone = 1;
And then don't need to check in a few places for 0 and the kbPerSecond is not 0.

Reply:
I changed it.

Reply:
But if you think its better or makes more sense checking for 0 and setting kbPerSecond and kbPerSecAvg to 0 then I will switch it.